### DISCUSSION TITLE
Firmware update and patching at same time

### DISCUSSION THREAD

**Discussion by CiberF** [Mr-MIBonk/M.I.B._More-Incredible-Bash#167] [Q&A] [answered]
Hi,
I have P2148 in A4 b9, and I want update it to 2161. After that I want to patch that I can install new Maps.

On mibwiki there is info that Patching only works on original FW. On the other hand if I understood well if I patch something and later install new Firmware everything patched will be deleted.

So, is there any way that I can both install new FW and patch it? Or I get something wrong. I searched wiki but have not found an answer.

Thanks and best wishes to everyone

**Comment by harman-f** [comment, marked as answer]
Sure, install. We FW and patch afterwards.

**Reply by CiberF** [reply]
Thank you very much.

**Comment by skamyra** [comment]
HI,
I don't know if this is the right post!!!!
But I wanted to ask you how can I delete SWDL Duplicates in my History FW update,because with For MIB2 HIGH see the jilleb/mib2-toolbox project
I don't understand how to do?
I add that I installed in my control unit MIB_More-Incredible-Bash.
Sorry for the inconvenience of the language, I'm Italian, thank you in advance.
![20230403_172906](https://user-images.githubusercontent.com/127048515/230032502-e2903ebd-4964-414d-afc7-48df9452b50e.jpg)
![20230403_172911](https://user-images.githubusercontent.com/127048515/230032747-b32c32db-b362-4833-bec9-85095c5369ad.jpg)
![GEM](https://user-images.githubusercontent.com/127048515/230032805-71d22b52-d306-4a61-b00c-9192d0e9aad8.png)
![20230403_172854](https://user-images.githubusercontent.com/127048515/230032862-771a1159-a60d-4ef4-a0bd-238520491559.jpg)

**Reply by harman-f** [reply]
History can be deleted, but currently there is no function available in M.I.B

**Comment by skamyra** [comment]
![Immagine 2023-04-05 110523](https://user-images.githubusercontent.com/127048515/230036062-fe16819f-a646-4e4b-95fd-84408e52f245.jpg)

**Reply by harman-f** [reply]
Not M.I.B
